### DISCUSSION TITLE
Stackoverflow in copyright license

### DISCUSSION THREAD

**Discussion by Franco0700** [oss-review-toolkit/ort#6580] [Q&A]
![image](https://user-images.githubusercontent.com/70175704/221962738-7be25a4b-2668-4126-839a-82918806dd68.png)
I'm running using docker with the following flags
`report -f PdfTemplate -i <input_file> -o <output_folder>`
I got that stackoverflow link in the copyright section of the license.
cc: @dgutson

**Comment by sschuberth** [comment]
You're not even asking a question, but I *guess* your question is how to get rid of that "false positive". There's currently [no way to curate Copyright findings](https://github.com/oss-review-toolkit/ort/issues/4519) in a similar way as license findings, so you'd have to go with the [Copyright garbage file](https://github.com/oss-review-toolkit/ort/blob/main/docs/config-file-copyright-garbage-yml.md).

**Reply by dgutson** [reply]
@sschuberth thanks for guiding us to the Copyright garbage file. Anyways maybe it would be interesting to analyze the root cause of this particular case and see if there is an opportunity for improving/fixing the (parser?) in case of a bug where a link to SO was obtained instead of the correct copyright? (@Franco0700 FWIW do you have the particular case where this came from, the dependency, I think it was mypy or something from pip?)

**Reply by sschuberth** [reply]
Well, here's the root cause analysis: There's no bug, the link to StackOverflow is literally used as a Copyright holder, see [here](https://github.com/Ousret/charset_normalizer/blob/5730a349b037312dd46c7541b7a3d62db7e1e985/charset_normalizer/cli/normalizer.py#L25). So it's completely valid for ScanCode (and ORT) to report it was a finding. If you as a user decide that you'd like to get rid of it, the Copyright garbage file is the way to go.

**Comment by dgutson** [comment]
Ah, it's right in the image: charset-normalizer.

**Comment by dgutson** [comment]
Sorry the spam. I just understood. The copyright is correct indeed: they are crediting a discussion in SO which led to or inspired the creation of this library. Arguably, the improvement could be a filter-list of this type (such as URLs of forums), what do you think @sschuberth ?

**Reply by sschuberth** [reply]
> Arguably, the improvement could be a filter-list of this type (such as URLs of forums), what do you think @sschuberth ?

I think that maintaining a list of valid forum URLs / patterns is a lot of work, and that the existing Copyright garbage mechanism already gets the job done. So I'd be 👎🏻 on such a quite specific feature.

If at all, I'd instead consider to extend the existing Copyright garbage mechanism to maybe work with prefixes, i.e. as soon as some configured string is a prefix of the encountered Copyright, it gets discarded as a whole.

**Reply by dgutson** [reply]
Does this mean that, with the current Copyright garbage mechanism, you can't say `*//stackoverflow.com/*` but instead
```
https://stackoverflow.com/questions/65050222/how-to-print-hello-world-in-python
```
?

**Reply by sschuberth** [reply]
That's correct. Not allowing wildcards was a deliberate decision to not unintendedly match something you did not think of.

**Reply by dgutson** [reply]
Could you please show how would look the stackoverflow filtering rule according to your proposal? Otherwise I can easily visualize the sad image of myself filtering every single link.

**Reply by sschuberth** [reply]
If you mean my proposal with the prefix, if that was implemented you'd need to determine a prefix that's common to all the URLs you'd like to treat as Copyright garbage, and add that to the garbage file. In your case probably a prefix of `https://stackoverflow.com/questions/`.
